File: request_api.py
import requests

BASE_URL = ''

def RequestEndpoint(image_url , key , score):       	
    URL = BASE_URL + '/predict?urls=' + image_url + '&score=' + str(score) + '&key=' + key
    r = requests.get(URL)
    data = r.json()
    return data['req_id'] 

def PercentageEndpoint(id):
    URL = BASE_URL + '/predict/percentage?req_id=' + id
    r = requests.get(URL)
    data = r.json()
    return data['percentage']

def OutputEndpoint(id):
    URL = BASE_URL + '/predict/output?req_id=' + id
    r = requests.get(URL)
    data = r.json()
    labels = ''
    for x in data['prediction']['output']:
        labels += x['id'] + " : " + x['label'] + "\n"
    return labels

if __name__== "__main__":
    
	key = "YOUR_KEY";
	image_url = "https://www.example.com/example_1.jpg,https://www.example.com/example_2.jpg";
	req_id = RequestEndpoint(image_url , key , 0);
	print(req_id);
	percentage = 0;
	while percentage != 100:
		percentage = PercentageEndpoint(req_id)
		print(percentage);	
	result = OutputEndpoint(req_id)
	print(result)

File: request_api.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using Json.Net;
using Newtonsoft.Json.Linq;

namespace API
{
    class Program
    {
        static string BASE_URL = "";
		
        private static async Task<string> RequestEndpoint(string image_url , string key , int score=0)
        {
            var url = BASE_URL + "/predict?urls=" + image_url + "&score=" + Convert.ToString(score) + "&key=" + key;
            using (var client = new HttpClient())
            {
                client.BaseAddress = new Uri(url);
                HttpResponseMessage response = await client.GetAsync(url);
                if (response.IsSuccessStatusCode)
                {
                    string strResult = await response.Content.ReadAsStringAsync();
                    JObject objects = JObject.Parse(strResult);
                    return (string)objects["req_id"]; 
                }
                else
                {
                    return null;
                }
            }
        }

        private static async Task<int> PercentageEndpoint(string id)
        {
            var url = BASE_URL +  "/predict/percentage?req_id=" + id;
            using (var client = new HttpClient())
            {
                client.BaseAddress = new Uri(url);
                HttpResponseMessage response = await client.GetAsync(url);
                if (response.IsSuccessStatusCode)
                {
                    string strResult = await response.Content.ReadAsStringAsync();
                    JObject objects = JObject.Parse(strResult);
                    return (int)objects["percentage"];
                }
                else
                {
                    return 0;
                }
            }
        }

        private static async Task<string> OutputEndpoint(string id)
        {
            var url = BASE_URL + "/predict/output?req_id=" + id;
            using (var client = new HttpClient())
            {
                client.BaseAddress = new Uri(url);
                HttpResponseMessage response = await client.GetAsync(url);
                if (response.IsSuccessStatusCode)
                {
                    string strResult = await response.Content.ReadAsStringAsync();
                    JObject objects = JObject.Parse(strResult);

                    string labels = "";
                    foreach(var x in objects["prediction"]["output"])
                    {
                        labels += (string)x["id"] + " : " + (string)x["label"] + "\n";
                    }

                    return labels;
                }
                else
                {
                    return null;
                }
            }
        }

        static void Main(string[] args)
        {

            string key = "YOUR_KEY";
            string image_url = "https://www.example.com/example_1.jpg,https://www.example.com/example_2.jpg";

            var req_id = RequestEndpoint(image_url , key).Result;
            Console.WriteLine(req_id);

            var percentage = 0;
            while (Convert.ToInt32(percentage) != 100)
            {
                percentage = PercentageEndpoint(req_id).Result;
                Console.WriteLine(percentage);
            }
            var result = OutputEndpoint(req_id).Result;
            Console.WriteLine(result);
        }
    }
}
